libbeat/processors/add_process_metadata/add_process_metadata_test.go

```diff
@@ -1118,8 +1126,293 @@ func TestV2CID(t *testing.T) {
 		}
 		return testMap, nil
 	}
-	provider := newCidProvider(resolve.NewTestResolver(""), []string{}, "", processCgroupPaths, nil)
+	provider := newCidProvider(resolve.NewTestResolver(""), nil, defaultCgroupRegex, processCgroupPaths, nil)
 	result, err := provider.GetCid(1)
 	assert.NoError(t, err)
 	assert.Equal(t, "2dcbab615aebfa9313feffc5cfdacd381543cfa04c6be3f39ac656e55ef34805", result)
 }
+
+func TestAddProcessMetadataWithV2(t *testing.T) {
```

Comment:
I think the new test cases should be incorporated into `TestAddProcessMetadata`. The distinction for cgroup v2 is not very meaningful.

And given that I created a test specifically for the `defaultCgroupRegex`, it's probably not necessary to add tests for every possible cgroup path format. I think the only test that is definitely necessary is "cgroup_prefixes configured" which checks for a breaking change.

Reply:
Yes. It was confusing for me too.